src/components/Repo.tsx

```diff
@@ -9,47 +11,87 @@ import { getBestContrastColor } from "@/lib/contrastColors";
 interface RepoProps {
   data: GithubData;
   key: number;
+  show: boolean;
 }
 
-const Repo: React.FC<RepoProps> = ({ data, key }) => {
+const Repo: React.FC<RepoProps> = ({ data, key, show }) => {
+  const [viewed, setViewed] = useState<boolean>(false);
+
+  useEffect(() => {
+    const storedRepos = localStorage.getItem("ttViewedRepos");
+
+    if (storedRepos) {
+      const repos = JSON.parse(storedRepos);
+      repos.forEach((repo: GithubData) => {
+        if (repo.id === data.id) {
+          setViewed(true);
+        }
+      });
+    } else {
+      setViewed(false);
+    }
+  }, [data]);
+
+  const handleOnClick = () => {
+    const existingRepos = localStorage.getItem("ttViewedRepos");
+    const repos = existingRepos ? JSON.parse(existingRepos) : [];
+
+    if (!repos.includes(data.id)) {
+      repos.push(data);
```

Comment:
also add the current date, so you can show when you have added the repo

Reply:
✅